docs/architecture/notes.md

```diff
@@ -196,10 +195,32 @@ hash(hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash), vault_hash)
 !!! info
     To compute a note's hash, we do not need to know the note's `serial_num`. Knowing the hash of the `serial_num` (as well as `script_hash`, `input_hash` and `note_vault`) is also sufficient. We compute the hash of `serial_num` as `hash(serial_num, [0; 4])` to simplify processing within the VM._
 
-## Note discovery
+## Note discovery (note tags)
+
+Note discovery describes the process by which Miden clients find notes they want to consume. Miden clients can query the Miden node for notes carrying a certain note tag. Note tags are best-effort filters for notes registered on the network. They are lightweight values used to speed up queries. Clients can follow tags for specific use cases, such as swap scripts, or agree with a group of users on certain tags for their specific needs. Tags are also used by the operator to identify notes intended for network execution and include the corresponding information on how to execute them.
+
+```arduino
+0b009f4adc47857e2f6
+```
```

Comment:
A couple of other things probably worth mentioning here:
- Note tags are 32-bit values (we say "lightweight" above - but I think it may be good to make it concrete).
- Note tags are a part of note metadata which is always public (not sure if we have a separate section about note metadata somewhere).

Reply:
done. 

We mention note metadata and explain that this is always public. But there is no separate section as we changed it to explain the note lifecycle rather than the note concepts.